engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/ExecutionStore.scala

```diff
@@ -44,94 +46,81 @@ case class ExecutionStore(private val store: Map[JobKey, ExecutionStatus]) {
       case node: GraphNode => node.upstreamAncestry exists hasFailedScope
     }
 
-    store exists {
-      case (jobKey, jobStatus) => (jobStatus == NotStarted && !upstreamFailed(jobKey.scope)) || jobStatus == QueuedInCromwell || jobStatus == Starting || jobStatus == Running
-    }
+    keysWithStatus(NotStarted).exists(jobKey => !upstreamFailed(jobKey.scope)) ||
+      keysWithStatus(QueuedInCromwell).nonEmpty ||
+      keysWithStatus(Starting).nonEmpty ||
+      keysWithStatus(Running).nonEmpty
```

Comment:
would this be more efficient to order the simple `nonEmpty` tests first before the `exists`?